### PR TITLE
Issue #12 homepage layout

### LINKED ISSUE #12: Homepage Layout
Based on the [Wireframing / Feature Planning](https://github.com/SE701-T5/Frontend/issues/6) issue:

- Code up the basic layout of the "Homepage" of the site.
- Colour scheme to be discussed in next team meeting (14/03/22)

### PR DESCRIPTION
**Pull request by yuchendong13**
Regarding [Issue #12](https://github.com/SE701-T5/Frontend/issues/12)

This PR includes the _basic_ layout of the home page as stated in issue 12. Placeholders exist where the Nav Bar and the "Popular Posts" feed will be (will be added after components are developed). Please let me know if I should hold off on a pull request for this and should wait until everything is put together before raising a PR. 

- Created PopularCommunity cards - temporarily populated the "Popular Communities" section with 3 static cards - this means that they will resize responsively on zoom in/out. This will be fixed once the horizontal scrolling is added.
- Set up containers for the homepage elements - the "Popular Communities" and "Popular Posts" containers exist
- Navbar and "Popular Posts" cards currently have placeholder text
- Colours subject to change - currently using the colour codes used on Figma for the font colour and background

Please pull the code and check if it looks ok on your devices.

## Type of change

- [x] New feature (enhancement)

## How Has This Been Tested?

- [x] Manual testing

# Checklist:

- [x] Does a similar (open or closed) pull request not already exist?
- [x] Is the pull request _head_ repository a fork repository?
- [x] Is the pull request _compare_ branch a development branch?
- [ ] Is the code documented, particularly in hard-to-understand areas?
- [x] Does the code build without new warnings?
- [x] Has testing been performed that proves changes are effective and work?
- [x] Has a self- and/or peer-review of the code been performed?
- [x] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [x] Is the person responsible for the repository assigned to the pull request?
- [x] Is the pull request linked to a project?
- [x] Is the pull request linked to a milestone?

### CODE REVIEW

**Review by christopher-alba — CHANGES_REQUESTED** [review]
image
Please resolve these unused vars issues so the deploy will succeed. If we merge into main the main deploy will fail too and we don't want that. Thanks

**Comment by yuchendong13** [comment]
Hi Chris, thanks for the review! I am definitely using the 'name,' 'alt,' and 'image' variables so I'm not sure what is going on. I'll have a closer look tomorrow and fix them.

File: src/components/PopularCommunity/PopularCommunity.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  
+    return (
+    <Card sx={{ maxWidth: 345 }} raised='true'>
+      <CardActionArea>
+      <Grid container spacing={1} direction='column'>
+        <Grid item>
+          <CardMedia
+            component="img"
+            height="140"
+            image={props.image}
```
Comment (christopher-alba) [inline src/components/PopularCommunity/PopularCommunity.jsx]:
```suggestion
            image={image}
```

File: src/components/PopularCommunity/PopularCommunity.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            alt={props.alt}
```
Comment (christopher-alba) [inline src/components/PopularCommunity/PopularCommunity.jsx]:
```suggestion
            alt={alt}
```

File: src/components/PopularCommunity/PopularCommunity.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          />
+        </Grid>
+        <Grid item>
+          <Typography variant="body">
+            {props.name}
```
Comment (christopher-alba) [inline src/components/PopularCommunity/PopularCommunity.jsx]:
```suggestion
            {name}
```

File: src/components/PopularCommunity/PopularCommunity.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Card from '@mui/material/Card';
+import { CardActionArea } from '@mui/material';
+import CardContent from '@mui/material/CardContent';
```
Comment (christopher-alba) [inline src/components/PopularCommunity/PopularCommunity.jsx]:
```suggestion
```

**Review by christopher-alba — CHANGES_REQUESTED** [review]
Please make these requested changes :) Mainly just using the deconstructed values from the props object. Will have to see what it looks like visually on the Netlify deploy once it succeeds. 
Thanks!

**Comment by yuchendong13** [comment]
Thanks Chris, I've made the changes and everything still looks the same. Hopefully netlify is happy now.

**Review by christopher-alba — CHANGES_REQUESTED** [review]
Its looking good! You dont have to be pixel perfect but please try and follow the figma as close as you can. This involves: 
- making the underlines of the titles larger and white
- making the text in the cards color #0080A7
- adding more spacing above the cards

**Review by christopher-alba — CHANGES_REQUESTED** [review]
Also another change we need to do is to append the name of the component before any class names to prevent the css from being applied to other pages. This also involves changing all css that targets elements directly to target classNames instead. For example: 
image
Instead of targeting img directly, apply a class name to the image and target that instead. Some names will be quite long so maybe just use the initials ( PopularCommunity will just be pc-image or something like that)

This is because even though we are using separate css files for each component, they all get bundled into one large css file in compile time so all the components will be affected by styles in any one css file. 
Thanks!

File: src/components/PopularCommunity/PopularCommunity.jsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import CardMedia from '@mui/material/CardMedia';
+import Typography from '@mui/material/Typography';
+import Grid from '@mui/material/Grid';
+import "./PopularCommunity.css";
+
+const PopularCommunity = (props) => {
+    const {name,alt,image}=props
+  
+    return (
+    <Card sx={{ maxWidth: 345 }} raised='true'>
```
Comment (hpar461) [inline src/components/PopularCommunity/PopularCommunity.jsx]: The code should be indented two spaces in following the indenting conventions of other files. This includes an indentation needed in line 13-32 (components inside `return ()` statement) and line 15-30 (components inside `<CardActionArea>` component)

File: src/pages/Homepage/Homepage.jsx [hunk 6]
```diff
@@ -1,8 +1,47 @@
...
+        </div>
+        <Container maxWidth="md">
+          Popular posts feed in this container
+        </Container>
+      </Container>
+      
+      
+    </>
+    
+
```
Comment (hpar461) [inline src/pages/Homepage/Homepage.jsx]: Redundant empty spaces here...? Or perhaps it has uses that I didn't understand.

**Review by hpar461 — COMMENTED** [review]
Great work!

File: src/components/PopularCommunity/PopularCommunity.jsx (hunk 5 above)
Comment (yuchendong13) [reply src/components/PopularCommunity/PopularCommunity.jsx]: Thanks for the feedback Hyung, the changes have been committed :)

File: src/pages/Homepage/Homepage.jsx (hunk 6 above)
Comment (yuchendong13) [reply src/pages/Homepage/Homepage.jsx]: You're right, my mistake - it's left over from my experimenting. All fixed!

**Review by hpar461 — DISMISSED** [review]

**Review by christopher-alba — DISMISSED** [review]
LGTM!

**Review by Xiaoqi510 — DISMISSED** [review]

**Review by hpar461 — DISMISSED** [review]
Great Work!

File: src/pages/Homepage/Homepage.jsx [hunk 7]
```diff
@@ -1,8 +1,47 @@
...
 import "./homepage.css";
+import Container from '@mui/material/Container';
+import PopularCommunity from "../../components/PopularCommunity/PopularCommunity.jsx";
+import Grid from '@mui/material/Grid';
 
 const Homepage = () => {
-  return <h1>Homepage</h1>;
+  return (
+    <div className="hpHomePage">
+      <h1 className="hpTitle">NavBar</h1> {/* Placeholder for Navbar component - to be replaced */}
```
Comment (christopher-alba) [inline src/pages/Homepage/Homepage.jsx]:
```suggestion
```

**Review by christopher-alba — CHANGES_REQUESTED** [review]
Nice work, just this one change.

**Review by christopher-alba — APPROVED** [review]
LGTM!

**Pull request merged**